sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+}
+
+// @public
+export type ListSparkSessionsResponse = SparkSessionCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSessionCollection;
+    };
+};
+
+// @public (undocumented)
+export type ListSparkStatementsOptions = OperationOptions;
+
+// @public
+export type ListSparkStatementsResponse = SparkStatementCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatementCollection;
+    };
+};
+
+// @public
+export const logger: import("@azure/logger").AzureLogger;
+
+// @public (undocumented)
+export interface OperationResponse {
+    _response: HttpResponse;
+}
+
+export { PipelineOptions }
+
+// @public
+export type PluginCurrentState = "Preparation" | "ResourceAcquisition" | "Queued" | "Submission" | "Monitoring" | "Cleanup" | "Ended" | string;
+
+// @public (undocumented)
+export type ResetSparkSessionTimeoutOptions = OperationOptions;
+
+// @public
+export type SchedulerCurrentState = "Queued" | "Scheduled" | "Ended" | string;
+
+// @public (undocumented)
+export interface SparkBatchJob {
+    appId?: string;
+    appInfo?: {
+        [propertyName: string]: string;
+    };
+    artifactId?: string;
+    errors?: SparkServiceError[];
+    id: number;
+    jobType?: SparkJobType;
+    // (undocumented)
+    livyInfo?: SparkBatchJobState;
+    logLines?: string[];
+    name?: string;
+    plugin?: SparkServicePlugin;
+    result?: SparkBatchJobResultType;
+    scheduler?: SparkScheduler;
+    sparkPoolName?: string;
+    state?: string;
+    submitterId?: string;
+    submitterName?: string;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    workspaceName?: string;
+}
+
+// @public
+export interface SparkBatchJobCollection {
+    from: number;
+    sessions?: SparkBatchJob[];
+    total: number;
+}
+
+// @public (undocumented)
+export interface SparkBatchJobOptions {
+    // (undocumented)
+    archives?: string[];
+    // (undocumented)
+    arguments?: string[];
+    // (undocumented)
+    artifactId?: string;
+    // (undocumented)
+    className?: string;
+    configuration?: {
+        [propertyName: string]: string;
+    };
+    // (undocumented)
+    driverCores?: number;
+    // (undocumented)
+    driverMemory?: string;
+    // (undocumented)
+    executorCores?: number;
+    // (undocumented)
+    executorCount?: number;
+    // (undocumented)
+    executorMemory?: string;
+    // (undocumented)
+    file: string;
+    // (undocumented)
+    files?: string[];
+    // (undocumented)
+    jars?: string[];
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    pythonFiles?: string[];
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export type SparkBatchJobResultType = "Uncertain" | "Succeeded" | "Failed" | "Cancelled" | string;
+
+// @public (undocumented)
+export interface SparkBatchJobState {
+    currentState?: string;
+    deadAt?: Date;
```

Comment:
the `at` suffix feels a little odd to me. @bterlson any thoughts here?

Reply:
Try as I might there's no real consistency here. My pedantic feeling is that `At` is appropriate when we're talking about a precise moment, and `On` is appropriate when we're talking about a date, but our type doesn't differentiate yet (our new Temporal proposal will allow for different suffixes to imply different types, though!). Moreover, in various cases we use `Date` and `Timestamp` suffixes.

I think `On` is most common though.